tensorly/tenalg/proximal.py

```diff
@@ -91,4 +91,316 @@ def procrustes(matrix):
+        V = tl.dot(scale, V)
+    else:
+        V = in_V
+
+    rho = 1
+    eps0 = 0
+    cnt = 1
+    eps = 1
+
+
+    while eps >= delta * eps0 and cnt <= 1 + 0.5* rho and cnt <= maxiter:
+        nodelta = 0
+        for k in range(r):
+
+            if UtU[k, k] != 0:
+                if sparsity_coefficient != None: # Modifying the objective function for sparsification
+                    if tl.get_backend() == 'pytorch':
+                        import torch
+                        deltaV = torch.maximum((UtM[k, :] - UtU[k, :] @ V - sparsity_coefficient * tl.ones(n)) / UtU[k, k],
+                                        -V[k, :])
+                    else:
+                        deltaV = tl.max([(UtM[k, :] - UtU[k, :] @ V - sparsity_coefficient * tl.ones(n)) / UtU[k, k],
+                                        -V[k, :]],axis=0)
+                    if tl.get_backend()=='tensorflow':
+                        import tensorflow as tf
+                        V=tf.Variable(V,dtype='float')
+                        V[k, :].assign(V[k, :] + deltaV)
+                    else:
+                        V[k, :] = V[k, :] + deltaV
+
+                else:  # without sparsity
+
+                    if tl.get_backend() == 'pytorch':
+                        import torch
+                        deltaV = torch.maximum((UtM[k, :] - tl.dot(UtU[k, :], V)) / UtU[k, k],
+                                         -V[k, :])
+                    else:
+                        deltaV = tl.max([(UtM[k, :] - tl.dot(UtU[k, :], V)) / UtU[k, k],
+                                         -V[k, :]], axis=0)
+                    if tl.get_backend()=='tensorflow':
+                        import tensorflow as tf
+                        V=tf.Variable(V,dtype='float')
+                        V[k, :].assign(V[k, :] + deltaV)
+                    else:
+                        V[k, :] = V[k, :] + deltaV
+
+                nodelta = nodelta + tl.dot(deltaV, tl.transpose(deltaV))
+
+                # Safety procedure, if columns aren't allow to be zero
+                if nonzero_row and (V[k, :] == 0).all():
+                    V[k, :] = 1e-16 * tl.max(V)
+
+            elif nonzero_row:
+                raise ValueError("Column " + str(k) + " of U is zero with nonzero condition")
+
+            if normalize:
+                norm = tl.norm(V[k,:])
+                if norm != 0:
+                    V[k,:] /= norm
+                else:
+                    sqrt_n = 1/n ** (1/2)
+                    V[k,:] = [sqrt_n for i in range(n)]
+        if cnt == 1:
+            eps0 = nodelta
+
+        rho_up=tl.shape(V)[0]*tl.shape(V)[1]+tl.shape(V)[1]*r
+        rho_down=tl.shape(V)[0]*r+tl.shape(V)[0]
+        rho=1+(rho_up/rho_down)
+        eps = nodelta
+        cnt += 1
+
+    return V, eps, cnt, rho
+def hals_nnls_exact(UtM, UtU, in_V, maxiter,delta=10e-12,sparsity_coefficient=None):
+
+    """
+    =================================
```

Comment:
We don't typically wrap the doc in a title, we just put the description straight after the quotes.